### DISCUSSION TITLE
Rendering dynamically-sized lists / arrays

### DISCUSSION THREAD

**Discussion by janoschherrmann** [amannn/next-intl#834] [Q&A] [answered]
I read the docs on [rendering arrays](https://next-intl-docs.vercel.app/docs/usage/messages#arrays-of-messages) in `next-intl`. For simple use cases, it works completely fine, but what if the messages are being generated dynamically by a CMS, and I want to allow variable length inputs? 

For example, say I want to have an FAQ component in my CMS, and want to be able to add new questions and answers as needed. In that case, using the approach described in the docs wouldn't work, as far as I can tell.

Is there a workaround for this?

**Comment by amannn** [comment, marked as answer]
> but what if the messages are being generated dynamically by a CMS, and I want to allow variable length inputs?

Can you provide an example of how the CMS output looks like?

From my experience a CMS might allow to add a custom FAQ _type_ (see [custom types in Prismic for example](https://prismic.io/docs/custom-types)) that can then be used on a content page to add multiple items, possibly depending on a locale ([locales in Prismic](https://prismic.io/docs/languages-locales)).

So from my understanding, `next-intl` shouldn't be necessary in this case.

On a related note, you can use rich text formatting to some degree for rendering lists of unknown size:

```tsx
"list": "<FaqItem><FaqTitle>First title</FaqTitle><FaqDescription>First description</FaqDescription></FaqItem><FaqItem><FaqTitle>Second title</FaqTitle><FaqDescription>Second description</FaqDescription></FaqItem>"
```

It's a bit tedious to edit for translators though, so a first-class feature in a CMS might be the better fit.

Hope this helps!

**Reply by janoschherrmann** [reply]
Ah, so when working with a CMS, it's generally recommended to use the localization from the CMS directly, instead of going through `next-intl`?

That makes a lot of sense actually, thanks!

**Reply by amannn** [reply]
I think for the content the CMS provides, absolutely—yes! `next-intl` can still be used in these apps for all labels that the CMS doesn't provide (i.e. anything that isn't strictly "content" and anything that needs dynamic arguments for example).

**Reply by janoschherrmann** [reply]
Got it, thank you for the quick help!
